composer/loggers/mosaicml_logger.py

```diff
@@ -96,6 +102,90 @@ def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
+
+        metrics: Dict[str, Any] = {'composer/autoresume': autoresume, 'composer/precision': trainer_state.precision}
+
+        train_dataloader = trainer_state.train_dataloader
+        if train_dataloader is not None and isinstance(train_dataloader, torch.utils.data.DataLoader):
+            metrics['composer/train_loader_workers'] = train_dataloader.num_workers
+
+        metrics['composer/eval_loaders'] = []
+        for evaluator in trainer_state.evaluators:
+            dataloader = evaluator.dataloader.dataloader
+            if isinstance(dataloader, torch.utils.data.DataLoader):
+                metrics['composer/eval_loaders'].append(
+                    json.dumps({
+                        'label': evaluator.label,
+                        'num_workers': dataloader.num_workers,
+                    }),
+                )
+
+        metrics['composer/optimizers'] = [
+            json.dumps(optimizer.state_dict(), sort_keys=True) for optimizer in trainer_state.optimizers
+        ]
+        metrics['composer/algorithms'] = [
+            json.dumps(algorithm.state_dict(), sort_keys=True) for algorithm in trainer_state.algorithms
+        ]
+        metrics['composer/loggers'] = [
+            get_logger_type(logger) if not isinstance(logger, MosaicMLLogger) else 'MosaicMLLogger'
+            for logger in loggers
+        ]
+
+        # Take the service provider out of the URI and log it to metadata. If no service provider
+        # is found (i.e. backend = ''), then we assume 'local' for the cloud provider.
+        if load_path is not None:
+            backend, _, _ = parse_uri(load_path)
+            metrics['composer/cloud_provider_data'] = backend if backend else 'local'
+        if save_folder is not None:
+            backend, _, _ = parse_uri(save_folder)
+            metrics['composer/cloud_provider_checkpoints'] = backend if backend else 'local'
+
+        # Save interval can be passed in w/ multiple types. If the type is a function, then
+        # we log 'callable' as the save_interval value for analytics.
+        if isinstance(save_interval, Union[str, int]):
+            save_interval_str = str(save_interval)
+        elif isinstance(save_interval, Time):
+            save_interval_str = f'{save_interval._value}{save_interval._unit}'
+        else:
+            save_interval_str = 'callable'
+        metrics['composer/save_interval'] = save_interval_str
+
+        if trainer_state.fsdp_config:
+            metrics['composer/sharding_strategy'] = trainer_state.fsdp_config.get('sharding_strategy', None)
+            metrics['composer/activation_checkpointing'] = trainer_state.fsdp_config.get(
+                'activation_checkpointing',
+                False,
+            )
+            metrics['composer/forward_prefetch'] = trainer_state.fsdp_config.get('forward_prefetch', False)
+            metrics['composer/backward_prefetch'] = trainer_state.fsdp_config.get('backward_prefetch', None)
```

Comment:
Looks like the default for backward prefetch is `BackwardPrefetch.BACKWARD_PRE`, see [here](https://pytorch.org/docs/stable/fsdp.html). Also, if torch changes these defaults then the logged metrics may be wrong. One idea is to initialize a `FullyShardedDataParallel` object on a dummy module with no args and retrieve the default values from that, maybe that'll work